services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/service/impl/NeighborService.java

```diff
@@ -140,10 +133,6 @@ public void buildNeighborStates(NetworkConfiguration networkConfig, String hostI
                         throw new NeighborInfoNotFound();
                     }
 
-                    if (hostIp.equals(neighborInfo.getHostIp())) {
-                        continue;
-                    }
```

Comment:
Will removal of this block put L2 neighbors on the same host in the GS message?

Reply:
I have included the checking for NeighborType.L3 if hostIps are the same to bypass the L2 neighbor_state in the same host.